plugins/services/src/js/reducers/serviceForm/common/Constraints.js

```diff
@@ -40,13 +45,44 @@ function processTransaction(state, { type, path, value }) {
 
 module.exports = {
   JSONParser(constraints) {
-    // Ignore non-array constraints
     if (!Array.isArray(constraints)) {
-      return [];
+      return [new Transaction(["constraints"], "not-list", ERROR)];
     }
 
     return constraints.reduce(function(memo, item, index) {
-      if (typeof item !== "object") {
+      if (Array.isArray(item)) {
+        memo.push(
+          new Transaction(
+            ["constraints", index, "value"],
+            "value-not-converted-to-object",
+            ERROR
+          )
+        );
+
+        return memo;
+      }
+
+      if (typeof item != "object") {
+        memo.push(
+          new Transaction(
+            ["constraints", index, "value"],
+            "value-not-object",
+            ERROR
+          )
+        );
+
+        return memo;
+      }
+
+      if (item.error) {
```

Comment:
<img width="1440" alt="screen shot 2018-01-29 at 11 26 07" src="https://user-images.githubusercontent.com/647035/35505712-409dce0a-04e7-11e8-81bb-4553f5cdb66c.png">

🎨 Please improve this check to avoid false positives (e.g. `{error:''}`).

Reply:
Done